AnimationCodelab/finished/src/main/java/com/example/android/codelab/animation/ui/home/Home.kt

```diff
@@ -0,0 +1,702 @@
+ *
+ * @param task The task description.
+ * @param onRemove Called when the task is swiped away and removed.
+ */
+@Composable
+private fun TaskRow(task: String, onRemove: () -> Unit) {
+    Surface(
+        modifier = Modifier
+            .fillMaxWidth()
+            .swipeToDismiss(onRemove),
+        elevation = 2.dp
+    ) {
+        Row(
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(16.dp)
+        ) {
+            Icon(
+                imageVector = Icons.Default.Check,
+                contentDescription = stringResource(R.string.task)
+            )
+            Spacer(modifier = Modifier.width(16.dp))
+            Text(
+                text = task,
+                style = MaterialTheme.typography.body1
+            )
+        }
+    }
+}
+
+/**
+ * The modified element can be horizontally swiped away.
+ *
+ * @param onDismissed Called when the element is swiped to the edge of the screen.
+ */
+private fun Modifier.swipeToDismiss(
+    onDismissed: () -> Unit
+): Modifier = composed {
+    val offsetX = remember { Animatable(0f) }
+    pointerInput(Unit) {
+        val decay = splineBasedDecay<Float>(this)
+        coroutineScope {
+            while (true) {
+                val pointerId = awaitPointerEventScope { awaitFirstDown().id }
+                val velocityTracker = VelocityTracker()
+                offsetX.stop()
+                awaitPointerEventScope {
+                    horizontalDrag(pointerId) { change ->
+                        launch {
+                            offsetX.snapTo(offsetX.value + change.positionChange().x)
+                        }
+                        velocityTracker.addPosition(change.uptimeMillis, change.position)
+                    }
+                }
+                val velocity = velocityTracker.calculateVelocity().x
+                val targetOffsetX = decay.calculateTargetValue(offsetX.value, velocity)
+                // The animation should end as soon as it reaches these bounds.
+                offsetX.updateBounds(
+                    lowerBound = -size.width.toFloat(),
+                    upperBound = size.width.toFloat()
+                )
+                launch {
+                    if (targetOffsetX.absoluteValue <= size.width) {
+                        // Not enough velocity; Slide back to the default position.
+                        offsetX.animateTo(targetValue = 0f, initialVelocity = velocity)
+                    } else {
+                        // Enough velocity to slide away the element to the edge.
+                        offsetX.animateDecay(velocity, decay)
+                        // The element was swiped away.
+                        onDismissed()
+                    }
+                }
+            }
+        }
+    }
+        .offset { IntOffset(offsetX.value.roundToInt(), 0) }
+}
+
+@Preview
+@Composable
+private fun PreviewHome() {
+    AnimationCodelabTheme {
+        Home()
+    }
+}
```

Comment:
Are there individual composables that might benefit from previews? Particularly any that might showcase inspection.

Reply:
Currently, I can't use Animation Inspector for updateTransition with `Int` state due to b/179512965.
I'll talk with the team to see if we can get the support soon or I should just modify the code to something that Animation Inspector can handle.

Reply:
shame :( Perhaps we can update if/when this is supported.